include/components/MovableCamera.hxx

```diff
@@ -0,0 +1,61 @@
+/*
+ *  MovableCamera.hxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef SMARTCAR_EMUL_MOVABLECAMERA_HXX
+#define SMARTCAR_EMUL_MOVABLECAMERA_HXX
+#include <unordered_map>
+#include <Urho3D/Core/Context.h>
+#include <Urho3D/Graphics/Camera.h>
+#include <Urho3D/Graphics/Viewport.h>
+#include <Urho3D/Scene/Component.h>
+#include <Urho3D/Scene/Scene.h>
+
+#define DEFAULT_MOVE_SPEED 20.0
+#define DEFAULT_MOVE_SENSITIVITY 0.1
+
+#define DEFAULT_FORWARD Urho3D::KEY_W
+#define DEFAULT_BACK Urho3D::KEY_S
+#define DEFAULT_LEFT Urho3D::KEY_A
+#define DEFAULT_RIGHT Urho3D::KEY_D
+
+class MovableCamera : public Urho3D::Camera {
+    URHO3D_OBJECT(MovableCamera, Camera);
+
+    float move_speed = DEFAULT_MOVE_SPEED;
+    float mouse_sensitivity = DEFAULT_MOVE_SENSITIVITY;
+
+    std::unordered_map<int, Urho3D::Vector3> key_map;
```

Comment:
While in code it looks fancy, you iterate through it every single tick, so please use a `std::vector<std::pair<?, Urho3D::Vector3>>`